graph/pull.go

```diff
@@ -265,6 +265,10 @@ func (s *TagStore) pullImage(r *registry.Session, out io.Writer, imgID, endpoint
 	// FIXME: Try to stream the images?
 	// FIXME: Launch the getRemoteImage() in goroutines
 
+	layers := append(history, imgID)
+	s.graph.Retain(layers, imgID)
+	defer s.graph.Unretain(history, imgID)
+
```

Comment:
hmm, it's really subtle here but I wanted to note that we `Retain` `[history..., imgID]` but later `Unretain` `history` only so that `imgID` has a remaining reference to itself. It's probably worth putting a comment here mentioning how important that is, if I'm understanding it correctly.